src/include/miopen/solver/implicitgemm_util.hpp

```diff
@@ -571,6 +569,13 @@ static inline auto get_static_ck_common_compiler_flag(const ExecutionContext& ct
     compiler_flag +=
         std::string(" -DCK_USE_AMD_V_FMAC_F32=") + (is_use_v_fmac_f32(ctx) ? '1' : '0');
 
+    compiler_flag += std::string(" -DCK_USE_AMD_BUFFER_PTR_TYPE=") +
+#if HIP_PACKAGE_VERSION_FLAT >= 6004000000
+                     '1'
+#else
+                     '0'
+#endif
+        ;
```

Comment:
```suggestion
    compiler_flag += " -DCK_USE_AMD_BUFFER_PTR_TYPE="
#if HIP_PACKAGE_VERSION_FLAT >= 6004000000
                     "1";
#else
                     "0";
#endif
```
The legacy of C will serve